src/MSLGenerator.cpp

```diff
@@ -222,6 +222,9 @@ void MSLGenerator::Prepass(HLSLTree* tree, Target target, HLSLFunction* entryFun
 
     HLSLType samplerType(HLSLBaseType_Sampler);
 
+    // Reserve buffer 0 for implicit vertex buffer
+    int bufferRegisterOffset = (m_target == Target_VertexShader) ? 1 : 0;
```

Comment:
Yes, I'm not very happy about this myself. Maybe it should be an offset you can pass to the generator and you'd be supposed to pass in 1 manually - I was also thinking for our code of removing the offset and trying to use buffers at a fixed offset for vertex streams (e.g. slots 8+ would be for vertex buffers...).